extensions/logs/suite.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logs
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+var logDirFlag = flag.String("log-dir", "logs", "container logs directory")
+var ignoreNamespaceFlag = flag.String("ignore-k8s-ns", "kube-system,local-path-storage",
+	"comma separated list of ignored kubernetes namespaces")
+
+type Suite struct {
+	suite.Suite
+
+	logDir        string
+	testStartTime time.Time
+	kubeClient    kubernetes.Interface
+}
+
+func (s *Suite) SetupSuite() {
+	s.logDir = *logDirFlag
+
+	kubeClient, err := newKubeClient()
+	require.NoError(s.T(), err)
+
+	s.kubeClient = kubeClient
+}
+
+func (s *Suite) SetupTest() {
+	s.testStartTime = time.Now()
+}
+
+func (s *Suite) AfterTest(suiteName, testName string) {
+	logPath := fmt.Sprintf("%s/%s/%s", s.logDir, suiteName, testName)
+	require.NoError(s.T(), os.MkdirAll(logPath, os.ModePerm))
+
+	logOptions := corev1.PodLogOptions{
+		Timestamps: true,
+		SinceTime:  &metav1.Time{Time: s.testStartTime},
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
Please move timeout for logging as a flag. Also, consider increase the timeout for 10-15 sec.

Reply:
Flag was added, default timeout was set to 15s